### DISCUSSION TITLE
Power Button Functionality (and Sleep Mode)

### DISCUSSION THREAD

**Discussion by no008080** [OnionUI/Onion#498] [Ideas]
I played around with your "competitor" MiniUI a bit, and i feel like it handles the power functions (and Frameskipping btw. :^) ) better than Onion as of the time of writing.

Wouldn't you think that hitting Power once should put it to sleep, while holding it should bring up the Power Off menu like on stock Miyoo FW?
Two people brought this up as an issue already.

On top of that, the automatic power down after two minutes of sleep, that MiniUI employs, seems like another great idea that would work really well with automatic save states.

I swear, i'm not some undercover agent here to advertise MiniUI, but i feel like both could learn from each other.
As of now Onion has the upper hand due to fixing the drivers thanks to Eggs, in my humble opinion. The devil is in the details, as they say.

**Comment by Wapp92** [comment]
I am one of those that hate that change. 
Best experience for me was. short press power button shutdown the screen and lock all keys, long press power complete shutdown ,menu+start save the game and exit.
I can understand why they did the changes. I would still love that they bring back the short press power (again the screen shutdown but the keys are lock, not to confuse with the sleep mode that is implemented Memu + power).
I hate the idea of automatic power down, but as a idea a customize Timer to shut down the device after xx:xx that would be cool.

**Comment by boxallw** [comment]
Yes. Miss being able to just turn off the screen.

**Comment by Sichroteph** [comment]
You can turn off the screen by pressing option + power. 
I actually tried MiniUI and I am deeply convinced that onion does it better in this regard. A device that boot in 8s needs more of a deep sleep than a light sleep. Just my opinion.

**Reply by Sichroteph** [reply]
Did you try the alternative keymon in apps ?
https://github.com/jimgraygit/Onion/wiki/6.-Miyoo-mini-apps

**Comment by thaatz** [comment]
I prefer how OnionOS works. It is faster to turn off and put away without needing to go through additional menus.

**Reply by boxallw** [reply]
Could not agree more. Super fast boot from off.
